### PR TITLE
[MIG] sale_layout_category_hide_detail: Migration to 15.0

### PR DESCRIPTION
**Pull request by ao-landoo**
Superseed #132

### CODE REVIEW

**Comment by ghost** [comment]
On a posted invoice when clicking on any button on the section it marks the invoice as paid when it shouldn't

**Review by ghost — APPROVED** [review]

**Comment by ao-landoo** [comment]
@victoralmau @elvise @francesco-ooops could you review this please?

**Comment by francesco-ooops** [comment]
@ao-landoo question: all these icons come from this module or some are from different modules?

image

**Comment by francesco-ooops** [comment]
@ao-landoo functional is ok, but what is this icon?

image

**Review by victoralmau — APPROVED** [review]
Code review OK

**Comment by rousseldenis** [comment]
> @ao-landoo functional is ok, but what is this icon?
> 
> image

@francesco-ooops This is to hide section subtotal

Done there: https://github.com/OCA/sale-reporting/commit/3b1313fecbadea781e5bd5dfc3b7e49038166be0

Forgot to update README.

@ao-landoo Could you add it in a separate commit in order to backport it ?

**Comment by francesco-ooops** [comment]
@rousseldenis can you also add in commit the readme part to add?

thanks!

**Comment by ao-landoo** [comment]
@rousseldenis Is it OK like this for you?

We could merge if you see it OK

**Comment by francesco-ooops** [comment]
@ao-landoo I would still add the description for what the letter icon is for (it's not really clear to me)

**Comment by ao-landoo** [comment]
> @ao-landoo I would still add the description for what the letter icon is for (it's not really clear to me)

How would you write the description? You can write it here and a can add it to Readme if it's OK

**Review by francesco-ooops — CHANGES_REQUESTED** [review]
@ao-landoo I'll try to describe the behavior, but on further inspection the flow looks a bit buggy and there's also an error on runboat  [video](https://recordit.co/edi8ajNo21)

File: sale_layout_category_hide_detail/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+This module allows you to check/uncheck the new 'Show details' and
+'Show subtotal' flags in *Sale order section lines* and
+*Customer invoice section lines* if you want to show or hide their
+lines or their subtotals in the PDF report and in the website.
+You can also show or hide the section subtotal by checking or
```
Comment (francesco-ooops) [inline sale_layout_category_hide_detail/readme/DESCRIPTION.rst]: You can also show or hide the section subtotal and line amount by checking or

File: sale_layout_category_hide_detail/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+unchecking the new 'Show section subtotal' flag.
```
Comment (francesco-ooops) [inline sale_layout_category_hide_detail/readme/DESCRIPTION.rst]: unchecking the new 'Show section subtotal' and "Show line amount" flags.

File: sale_layout_category_hide_detail/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+following:
+
+  * The first 'line section' and its product order lines will be shown in
+    a normal way.
+  * The second 'line section' and its product order lines will be shown in
+    a normal way, but the subtotal won't be shown. That is because in this
+    section line *Show subtotal* field was set to False.
+  * The third 'line section' will show the name on the left and the
+    subtotal on the right. Besides, its product order lines won't be shown.
+    That is because in this line *Show details* field was set to False.
```
Comment (francesco-ooops) [inline sale_layout_category_hide_detail/readme/USAGE.rst]: The flags on the right hand side of the sale order line allow you to show or hide details, subtotal, line amount and section subtotal.

These are the possible combinations: 

image


And these are the results in report: 

image

**Review by francesco-ooops — CHANGES_REQUESTED** [review]
This is how I would change description and usage

File: sale_layout_category_hide_detail/readme/USAGE.rst (hunk 3 above)
Comment (ao-landoo) [reply sale_layout_category_hide_detail/readme/USAGE.rst]: Would you replace all the previous content for the content you wrote here?

Comment (francesco-ooops) [reply sale_layout_category_hide_detail/readme/USAGE.rst]: yes exactly, takes too long to describe every combination - all you need to know is that there are icons and when they're toggled you get those results

Comment (ao-landoo) [reply sale_layout_category_hide_detail/readme/USAGE.rst]: Done, I'm not completely sure that images are correcly added. Could you check?

Comment (francesco-ooops) [reply sale_layout_category_hide_detail/readme/USAGE.rst]: The description is ok, let me know when I can test functionally again

Comment (ao-landoo) [reply sale_layout_category_hide_detail/readme/USAGE.rst]: It's ready for functional review

**Comment by francesco-ooops** [comment]
> @ao-landoo I'll try to describe the behavior, but on further inspection the flow looks a bit buggy and there's also an error on runboat [video](https://recordit.co/edi8ajNo21)

@ao-landoo this was not addressed

ps: the "buggy flow" refers to the fact that as you see at 1:02 and 1.07 in the video, the "letter" icon is appearing only when clicking another icon on another section, but that was happening also in v14 so maybe no need to fix it now

**Comment by ao-landoo** [comment]
> > @ao-landoo I'll try to describe the behavior, but on further inspection the flow looks a bit buggy and there's also an error on runboat [video](https://recordit.co/edi8ajNo21)
> 
> @ao-landoo this was not addressed
> 
> ps: the "buggy flow" refers to the fact that as you see at 1:02 and 1.07 in the video, the "letter" icon is appearing only when clicking another icon on another section, but that was happening also in v14 so maybe no need to fix it now

As you said, this also happens in v14 so it's not part of the migration

**Comment by francesco-ooops** [comment]
@ao-landoo yes, but in v14 the error appearing at 0.38 is not present, I think that should be fixed

**Comment by ao-landoo** [comment]
> @ao-landoo yes, but in v14 the error appearing at 0.38 is not present, I think that should be fixed

Cannot find the origin of the problem.. Looks like something changed on javascript from versión 14.0 but I'm unable to find the solution

**Comment by victoralmau** [comment]
Please, cherry-pick https://github.com/OCA/sale-reporting/pull/181 to commit history

**Comment by ao-landoo** [comment]
@francesco-ooops Can you confirm the traceback at 0:38 in your video is now fixed?

**Review by francesco-ooops — APPROVED** [review]
functional ok!

**Comment by ao-landoo** [comment]
@pedrobaeza Can it be merged?

**Pull request merged**